backend/api/routes/jobs.py

```
@@ -523,6 +524,311 @@ async def get_instrumental_options(job_id: str) -> Dict[str, Any]:
+            num_points=num_points,
+        )
+        
+        return {
+            "amplitudes": amplitudes,
+            "duration_seconds": duration,
+            "num_points": len(amplitudes),
+        }
+        
+    except Exception as e:
+        logger.error(f"Error getting waveform data for job {job_id}: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)